HCCGo/app/js/jobViewCtrl.js

```diff
@@ -75,4 +75,46 @@ jobViewModule.controller('jobViewCtrl', ['$scope', '$log', '$timeout', 'connecti
     });
   });
 
+  $scope.saveFile = function(fileType, $event) {
+    const {dialog} = require('electron').remote;
+    const fs = require('fs');
+    var options = {
+      filters: [
+        { name: 'text', extensions: ['txt']}
+      ]
+    };
+
+    dialog.showSaveDialog(options, function(localFile) {
+      // If the user clicks cancel, localFile will be undefined
+      if (!localFile) return;
+      switch(fileType){
+        case 'Output':
+          if (!downloadRemoteFile($scope.job.outputPath, localFile))
+            fs.writeFile(localFile, $scope.job.outText, function(err) {});
+          break;
+        case 'Error':
+          if (!downloadRemoteFile($scope.job.errorPath, localFile))
+            fs.writeFile(localFile, $scope.job.errText, function(err) {});
+          break;
+      }
+    });
+    // Stop the propagation of the click
+    $event.stopPropagation();
+  };
+
+  function downloadRemoteFile(remotePath, localPath) {
+    connectionService.getFileSize(remotePath).then(function(size) {
+      if (size > 5*1025*1024) {
+        connectionService.quickDownload(remotePath, localPath);
+        return true;
+      }
+      return false;
+    });
+  }
+
+  $scope.copyToClipboard = function(fileType, $event) {
```

Comment:
I like this logic of using the `fileType`, why don't you do that above instead of the switch/case statement?